src/main/java/seedu/task/model/task/Event.java

```diff
@@ -57,6 +58,16 @@ public boolean equals(Object other) {
                 && otherTask.getTo().equals(getTo());
     }
 
+    /**
+     * Returns true if {@code to} is within the {@code alertWindow} from the current time.
+     */
+    @Override
+    public boolean isComingUp() {
+        boolean isAfterNow = this.to.value.isAfter(LocalDateTime.now());
+        boolean isBeforeAlert = this.to.value.minus(alertWindow).isBefore(LocalDateTime.now());
+        return isAfterNow && isBeforeAlert;
```

Comment:
I would like to suggest checking for an event "from" time as well. An example use case would be an event from 6pm to 8pm. A user checks the app at 5pm with "alert 2" and will be able to see that there is a dinner starting soon.

Reply:
Great suggestion! I will add that check.